x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/CommandLineHttpClient.java

```diff
@@ -318,7 +318,7 @@ public static HttpResponse.HttpResponseBuilder responseBuilder(InputStream is) t
     }
 
     public static URL createURL(URL url, String path, String query) throws MalformedURLException, URISyntaxException {
-        return new URL(url, (url.toURI().getPath() + path).replaceAll("/+", "/") + query);
+        return new URL(url, (url.toURI().getPath() + path).replaceAll("//+", "/") + query);
```

Comment:
This change makes sense to me: replacing a single `/` with `/` is a useless and probably inefficient operation, so we should only trigger a replacement for two or more forward slashes. There are enough cases of this that I think you should mention it in the PR description.